tests/kernel/interrupt/src/dynamic_isr.c

```diff
@@ -50,10 +60,74 @@ void test_isr_dynamic(void)
 		     _sw_isr_table[i].arg == argval,
 		     "dynamic isr did not install successfully");
 }
+
+#else
+/*
+ * For testing arch such as x86, x86_64 and posix which support dynamic
+ * interrupt but without SW ISR table, we test it by applying for a dynamic
+ * interrupt and then trigger it to check if happened correctly.
+ */
+#if defined(CONFIG_ARCH_POSIX)
+#include "irq_ctrl.h"
+#endif
+
+static int get_dynamic_interrupt_line(void)
+{
+#if defined(CONFIG_X86)
+	return TEST_IRQ_DYN_LINE;
+
+#elif defined(CONFIG_ARCH_POSIX)
+	hw_irq_ctrl_enable_irq(TEST_IRQ_DYN_LINE);
+	return TEST_IRQ_DYN_LINE;
+#endif
+}
+
+void test_isr_dynamic(void)
+{
+	int irq_dyn_line;
+	int vector_num;
+
+	/* Get a irq line for dynamic interrupt */
+	irq_dyn_line = get_dynamic_interrupt_line();
+
+	/**TESTPOINT: configuration of interrupts dynamically at runtime */
+	vector_num = arch_irq_connect_dynamic(irq_dyn_line, 1, dyn_isr,
+				 (void *)ISR_DYN_ARG, 0);
+
+#if defined(CONFIG_X86_64)
+/* The isr table for x86_64 is visiable, so check it up here */
+extern void (*x86_irq_funcs[])(const void *);
+extern const void *x86_irq_args[];
+
+	zassert_true(x86_irq_funcs[irq_dyn_line] == dyn_isr &&
+		     x86_irq_args[irq_dyn_line] == (void *)ISR_DYN_ARG,
+		     "dynamic isr did not install successfully");
+#endif
+
+	TC_PRINT("vector(%d)\n", vector_num);
+	zassert_true(vector_num > 0,
+			"irq connect dynamic failed");
+
+	zassert_equal(handler_has_run, 0,
+			"(%d)should not get parameter before interrupt",
```

Comment:
This is about interrupt being triggered but should not be. So maybe something like `handler has run before interrupt trigger` which is more descriptive.

Reply:
Yes, the words I used here is not adequate, just got it fix, thanks~